com.unity.ml-agents/Runtime/Agent.cs

```diff
@@ -362,36 +364,49 @@ void NotifyAgentDone(DoneReason doneReason)
         /// Updates the Model for the agent. Any model currently assigned to the
         /// agent will be replaced with the provided one. If the arguments are
         /// identical to the current parameters of the agent, the model will
-        /// remain unchanged.
+        /// remain unchanged, unless the force parameter is true.
         /// </summary>
         /// <param name="behaviorName"> The identifier of the behavior. This
         /// will categorize the agent when training.
         /// </param>
         /// <param name="model"> The model to use for inference.</param>
         /// <param name = "inferenceDevice"> Define on what device the model
         /// will be run.</param>
+        /// <param name="force">Whether to update the Agent even if all the parameters are the same.</param>
         public void GiveModel(
             string behaviorName,
             NNModel model,
-            InferenceDevice inferenceDevice = InferenceDevice.CPU)
+            InferenceDevice inferenceDevice = InferenceDevice.CPU,
+            bool force = false)
```

Comment:
This is a bit convoluted; we don't want to make a new policy if the arguments are the same as the behavior parameters. *BUT* if we're coming from the BehaviorParametersEditor, we know those fields changed from the old values, and that they also match the current BehaviorParameters's values. Thus we need to "force" the update.

Reply:
I don't understand why we need to force the update of the Agent's Policy if the model did not change. Doesn't that only correspond to re-dragging the same model into the `Model` field of the inspector or am I missing something?

Reply:
1) If you change the BehaviorName or InferenceDevice, we also need to update the policy (at least I'm pretty sure, please correct me if that's wrong)

2) By the time we get to calling this method from here
https://github.com/Unity-Technologies/ml-agents/blob/477d5cb5e2f45dcdaf69cdb57aab0fee393773ca/com.unity.ml-agents/Editor/BehaviorParametersEditor.cs#L134-L139
The BehaviorParameters were already updated by 
https://github.com/Unity-Technologies/ml-agents/blob/477d5cb5e2f45dcdaf69cdb57aab0fee393773ca/com.unity.ml-agents/Editor/BehaviorParametersEditor.cs#L70
so unless we have a `force` option, the update would always be optimized away.

Reply:
Ignore the above comment; ReloadPolicy() suggested below makes this irrelvant.